### ISSUE TITLE
Error: SqlxError# error returned from database: (code: 5) database is locked

### ISSUE THREAD

**Issue by manoj-malviya** [openobserve/openobserve#1549]
### Which OpenObserve functionalities are the source of the bug?

ingestion

### Is this a regression?

Yes

### Description

Observe stop responding when many logs are pushed.

I am running Openobserve in docker in AWS EC2 machine (t4g.large) .  OS is ubuntu

Following is my docker-compose.yml

```
version: '3.7'

services:
  open-observe:
    container_name: open-observe
    image: public.ecr.aws/zinclabs/openobserve:latest-arm64
    volumes:
      - /open-data:/data
    ports:
      - 5080:5080
    environment:
      - ZO_DATA_DIR=/data
      - ZO_ROOT_USER_EMAIL=*
      - ZO_ROOT_USER_PASSWORD=*
      - ZO_DATA_LIFECYCLE=21
```

```
[2023-09-20T18:10:55Z INFO  openobserve::service::schema] service:schema:schema_evolution; org_id="default" stream_name="b2c-app" stream_type=Logs min_ts=1695233217307304
[2023-09-20T18:10:55Z INFO  openobserve::job::files::disk] [JOB] disk file upload succeeded: files/default/logs/b2c-app/2023/09/20/18/7110323494109446150pNpwcC.parquet
[2023-09-20T18:11:00Z ERROR openobserve::service::db::file_list] service:db:file_list: add files/default/logs/b2c-app/2023/09/20/18/7110323494109446146Kpyetu.parquet, set_file_to_cache error: Error# (code: 5) database is locked
[2023-09-20T18:11:00Z INFO  openobserve::job::files::disk] [JOB] convert disk file: /data/wal/files/default/logs/b2c-app/0/2023/09/20/18/90f67e86316a71fd/7110323494109446149bnIiQB.json
[2023-09-20T18:11:00Z INFO  openobserve::job::files::disk] [JOB] File upload begin: disk: /data/wal/files/default/logs/b2c-app/0/2023/09/20/18/894ecc206e27b7fd/7110323494109446147b6sgwt.json
[2023-09-20T18:11:00Z INFO  openobserve::service::schema] service:schema:schema_evolution; org_id="default" stream_name="b2c-app" stream_type=Logs min_ts=1695233213917998
[2023-09-20T18:11:00Z INFO  openobserve::job::files::disk] [JOB] disk file upload succeeded: files/default/logs/b2c-app/2023/09/20/18/7110323494109446147b6sgwt.parquet
[2023-09-20T18:11:05Z ERROR openobserve::service::db::file_list] service:db:file_list: add files/default/logs/b2c-app/2023/09/20/18/7110323494109446150pNpwcC.parquet, set_file_to_cache error: Error# (code: 5) database is locked
[2023-09-20T18:11:05Z INFO  openobserve::job::files::disk] [JOB] convert disk file: /data/wal/files/default/logs/b2c-app/0/2023/09/20/18/a3d744f771073275/7110323494109446151iEbE3N.json
[2023-09-20T18:11:05Z INFO  openobserve::job::files::disk] [JOB] File upload begin: disk: /data/wal/files/default/logs/b2c-app/0/2023/09/20/18/90f67e86316a71fd/7110323494109446149bnIiQB.json
[2023-09-20T18:11:05Z INFO  openobserve::service::schema] service:schema:schema_evolution; org_id="default" stream_name="b2c-app" stream_type=Logs min_ts=1695233213700583
[2023-09-20T18:11:05Z INFO  openobserve::job::files::disk] [JOB] disk file upload succeeded: files/default/logs/b2c-app/2023/09/20/18/7110323494109446149bnIiQB.parquet
[2023-09-20T18:11:05Z ERROR openobserve::service::db::file_list] service:db:file_list: add files/default/logs/b2c-app/2023/09/20/18/7110323494109446147b6sgwt.parquet, set_file_to_cache error: Error# (code: 5) database is locked
[2023-09-20T18:11:05Z INFO  openobserve::job::files::disk] [JOB] File upload begin: disk: /data/wal/files/default/logs/b2c-app/0/2023/09/20/18/a3d744f771073275/7110323494109446151iEbE3N.json
[2023-09-20T18:11:05Z INFO  openobserve::service::schema] service:schema:schema_evolution; org_id="default" stream_name="b2c-app" stream_type=Logs min_ts=1695233216651106
[2023-09-20T18:11:05Z INFO  openobserve::job::files::disk] [JOB] disk file upload succeeded: files/default/logs/b2c-app/2023/09/20/18/7110323494109446151iEbE3N.parquet
[2023-09-20T18:11:10Z ERROR openobserve::service::db::file_list] service:db:file_list: add files/default/logs/b2c-app/2023/09/20/18/7110323494109446149bnIiQB.parquet, set_file_to_cache error: Error# (code: 5) database is locked
[2023-09-20T18:11:15Z ERROR openobserve::service::db::file_list] service:db:file_list: add files/default/logs/b2c-app/2023/09/20/18/7110323494109446151iEbE3N.parquet, set_file_to_cache error: Error# (code: 5) database is locked
[2023-09-20T18:11:35Z ERROR openobserve::job::compact] [COMPACTOR] run data merge error: SqlxError# error returned from database: (code: 5) database is locked


[2023-09-20T18:12:35Z ERROR openobserve::job::compact] [COMPACTOR] run data merge error: SqlxError# error returned from database: (code: 5) database is locked
[2023-09-20T18:13:35Z ERROR openobserve::job::compact] [COMPACTOR] run data merge error: SqlxError# error returned from database: (code: 5) database is locked
[2023-09-20T18:14:35Z ERROR openobserve::job::compact] [COMPACTOR] run data merge error: SqlxError# error returned from database: (code: 5) database is locked
[2023-09-20T18:15:35Z ERROR openobserve::job::compact] [COMPACTOR] run data merge error: SqlxError# error returned from database: (code: 5) database is locked
[2023-09-20T18:16:35Z ERROR openobserve::job::compact] [COMPACTOR] run data merge error: SqlxError# error returned from database: (code: 5) database is locked
```

**Comment by prabhatsharma** [comment]
How much data are you pushing? MB/second? For sustained high throughput t family of instances is not the right fit. With t4g.large instance, you get only 30% of the CPU as baseline (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/burstable-credits-baseline-concepts.html) . You do get additional credits, but you lose them if you keep using the instance at full capacity. I would recommend using c7g/m7g instances for a better understanding of what capacity you need for your workload.

OpenObserve is much more efficient than elasticsearch but it still needs some baseline hardware. You can ingest 7-20 MB logs per second per vCPU as baseline. On t4g.large with sustained throughput, you should be able to ingest 4-12 MB/second.

**Comment by hengfeiyang** [comment]
What is your resource limit? CPU and Memory

**Comment by manoj-malviya** [comment]
CPU is not the blocker here, I can see CPU is almost idle.
Ingest logs are way smaller than 4mb/second.

I have stopped log ingestion but logs still showing "Database is locked" and Web UI is non-responsive.

I will try with other instance type and see if it improves.

**Comment by hengfeiyang** [comment]
Can you try to use a node that has 4 CPU cores.

**Comment by manoj-malviya** [comment]
I tried on m7g.xlarge instance.

It worked for some time then again started giving error. At the same time UI become un-responsive.

Here is the log file if it helps.

https://drive.google.com/file/d/13JWIcDytFeJcXssn_Ap0ZTFGiRi7rnnN/view?usp=drive_link

**Comment by martadinata666** [comment]
I don't know how your `/open-data` directory is set up, but database locked is a known limit of sqlite if the directory is mounted remotely such as `NFS` and alike.

**Comment by hengfeiyang** [comment]
We found this is a bug, we are fixing the error `database is locked`

**Comment by hengfeiyang** [comment]
fixed in `v0.6.3`
